### ISSUE TITLE
Column visibility doesn't work if data loading from server

### ISSUE THREAD

**Issue by maslianok** [DataTables/Buttons#8]
Column visibility control will not work with next config:
```javascript
$('#table').DataTable({
	"serverSide":true,
	"sAjaxSource":"http://localhost/users",
	"dom":"<\"dataTables_header\"Blfr>t<\"dataTables_footer\"ip>",
	"columns":[{
		"aTargets":"name",
		"mDataProp":"name",
		"title":"Name",
		"visible":true
	}, {
		"aTargets":"position",
		"mDataProp":"position",
		"title":"Position",
		"visible":true
	}],
	"buttons":[{"extend":"colvis","className":"button"}],
	"aaSorting":[[0,"asc"]]
});
```
You will get an empty box instead of column names
![error](http://i.imgur.com/YEQ1Ecm.png) 

I've changed columnVisibility method in buttons.colVis.js:
```javascript
//before
columnVisibility: {
	/*...*/
	text: function ( dt, button, conf ) {
		return $(dt.column( conf.columns ).header()).text();
	},
	/*...*/
},
//after
columnVisibility: {
	/*...*/
	text: function ( dt, button, conf ) {
		return dt.settings()[0].oInit.columns[conf.columns].title;
	},
	/*...*/
},
```
and it work fine for me. But I suppose that it will not work for usual tables. Maybe it should looks like:
```javascript
columnVisibility: {
	/*...*/
	text: function ( dt, button, conf ) {
		return dt.settings()[0].sAjaxSource ? 
			dt.settings()[0].oInit.columns[conf.columns].title :
			$(dt.column( conf.columns ).header()).text();
	},
	/*...*/
},
```

**Comment by DataTables** [comment]
Thanks. I'll take a look in the next few days.

**Comment by DataTables** [comment]
Fixed by commit 5f780c773af9e1cf6. Thanks for letting me know about this!
